### PR TITLE
[SYSTEMDS-2925] Federated left indexing

### PR DESCRIPTION
**Pull request by OlgaOvcharenko**
This PR adds federated left indexing, the simplest case when rhs matrix or frame is read and broadcast sliced (e.g. A[2:12, 3:4] = B).

### CODE REVIEW

File: src/main/java/org/apache/sysds/runtime/instructions/fed/IndexingFEDInstruction.java [hunk 1]
```diff
@@ -171,4 +190,110 @@ private void rightIndexing(ExecutionContext ec)
...
+			long cen = (ixrange.colEnd >= cs && ixrange.colEnd < ce) ? (ixrange.colEnd - cs) : (ce - cs - 1);
+
+			long[] newIx = new long[]{(int) rsn, (int) ren, (int) csn, (int) cen};
+
+			// find ranges where to apply  leftIndex
+			long to;
+			if(in1.isFederated(FederationMap.FType.ROW) && (to = (prev + ren - rsn)) >= 0 &&
+					to < in2.getNumRows() && ixrange.rowStart <= re) {
+					sliceIxs[i] = new int[] { prev, (int) to, 0, (int) in2.getNumColumns()-1};
+					prev = (int) (to + 1);
```
Comment (Baunsgaard) [inline src/main/java/org/apache/sysds/runtime/instructions/fed/IndexingFEDInstruction.java:237]: something has gone wrong with the indentation

File: src/main/java/org/apache/sysds/runtime/instructions/fed/IndexingFEDInstruction.java [hunk 2]
```diff
@@ -171,4 +190,110 @@ private void rightIndexing(ExecutionContext ec)
...
+					from = Math.min(i, from);
+			} else if(in1.isFederated(FederationMap.FType.COL) && (to = (prev + cen - csn)) >= 0 &&
+					to < in2.getNumColumns() && ixrange.colStart <= ce) {
+					sliceIxs[i] = new int[] {0, (int) in2.getNumRows() - 1, prev, (int) to};
+					prev = (int) (to + 1);
+
+					instStrings[i] = modifyIndices(newIx, 4, 8);
+					ranges[i] = range;
+					from = Math.min(i, from);
+			} else
```
Comment (Baunsgaard) [inline src/main/java/org/apache/sysds/runtime/instructions/fed/IndexingFEDInstruction.java:250]: code style specifies newline before else after a }

**Review by Baunsgaard — COMMENTED** [review]
LGTM, except a bit of formatting

File: src/main/java/org/apache/sysds/runtime/instructions/fed/IndexingFEDInstruction.java (hunk 2 above)
Comment (OlgaOvcharenko) [reply src/main/java/org/apache/sysds/runtime/instructions/fed/IndexingFEDInstruction.java]: Thanks for the review, fixed the formatting now

**Pull request closed without merging**